src/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -24,6 +24,62 @@ private struct DateTimeParseData
             public byte OffsetToken;
         }
 
+        public static string FormatDateTimeOffset(DateTimeOffset value)
+        {
+            Span<byte> span = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            JsonWriterHelper.WriteDateTimeOffsetTrimmed(span, value, out int bytesWritten);
+
+            return JsonReaderHelper.GetTextFromUtf8(span.Slice(0, bytesWritten));
+        }
+
+        public static string FormatDateTime(DateTime value)
+        {
+            Span<byte> span = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
```

Comment:
```suggestion
            Span<byte> span = stackalloc byte[JsonConstants.MaximumFormatDateTimeLength];
```

Reply:
Yeah, I wondered that as well. I juts kept the code as-is. Might've been a bug 🤷🏻‍♂️

Reply:
The constant used is correct. `Utf8Formatter` will write either 27, 28, or 33 bytes as a `DateTime` format "O" representation:

```C#
using System;
using System.Buffers;
using System.Buffers.Text;
using System.Text;
					
public class Program
{
	public static void Main()
	{
		// Utc offset info.
		Span<byte> buffer1 = new byte[33];
		Utf8Formatter.TryFormat(DateTime.UtcNow, buffer1, out int bytesWritten, new StandardFormat('O'));
		Console.WriteLine(Encoding.UTF8.GetString(buffer1));
		Console.WriteLine(bytesWritten);
		
		// Local offset info.
		Span<byte> buffer2 = new byte[33];
		Utf8Formatter.TryFormat(DateTime.Now, buffer2, out bytesWritten, new StandardFormat('O'));
		Console.WriteLine(Encoding.UTF8.GetString(buffer2));
		Console.WriteLine(bytesWritten);
		
		// No offset info.
		Span<byte> buffer3 = new byte[33];
		Utf8Formatter.TryFormat(new DateTime(2019, 5, 12), buffer3, out bytesWritten, new StandardFormat('O'));
		Console.WriteLine(Encoding.UTF8.GetString(buffer3));
		Console.WriteLine(bytesWritten);
	}
}

// Output:
// 2019-10-26T02:09:08.3810181Z
// 28
// 2019-10-26T02:09:08.3873201+00:00
// 33
// 2019-05-12T00:00:00.0000000
// 27
```

The output of writing `DateTimeOffset` will always be 33 bytes (expressed as a DateTime with non-UTC offset information, e.g. `2019-10-26T02:21:10.9689815+00:00`).

For clarity,
- a comment explaining these semantics should be added above this allocation,
- `JsonConstants.MaximumFormatDateTimeLength` should be renamed to `MinimumFormattedDateTimeLength`, and
- `JsonConstants.MaximumFormatDateTimeOffsetLength` should be renamed to `FormattedDateTimeOffsetLength` (or something clearer).